### PR TITLE
hostinfo, ipnlocal: flesh out Wake-on-LAN support, send MACs, add c2n sender

### PR DESCRIPTION
**Pull request by bradfitz**
This optionally uploads MAC address(es) to control, then adds a
c2n handler so control can ask a node to send a WoL packet.

Updates #306

### CODE REVIEW

File: hostinfo/wol.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+// TODO(bradfitz): this is all too simplistic and static. It needs to run
+// continuously in response to netmon events (USB ethernet adapaters might get
+// plugged in) and look for the media type/status/etc. Right now on macOS it
+// still detects a half dozen "up" en0, en1, en2, en3 etc interfaces that don't
+// have any media. We should only report the one that's actually connected.
+// But it works for now (2023-10-05) for fleshing out the rest.
+
+var wakeMAC = envknob.RegisterString("TS_WAKE_MAC") // mac address, "false" or "auto". for https://github.com/tailscale/tailscale/issues/306
+
+func getWoLMACs() (macs []string) {
```
Comment (maisem) [inline hostinfo/wol.go:25]: add some docs describing what it returns in what platforms, and that auto is limited to 10

File: ipn/ipnlocal/c2n.go [hunk 2]
```diff
@@ -270,3 +277,56 @@ func findCmdTailscale() (string, error) {
...
+	st := b.sys.NetMon.Get().InterfaceState()
+	if st == nil {
+		res.Errors = append(res.Errors, "no interface state")
+		writeJSON(w, &res)
+		return
+	}
+	var password []byte // TODO(bradfitz): support? does anything use WoL passwords?
+	for _, mac := range macs {
+		for ifName, ips := range st.InterfaceIPs {
+			for _, ip := range ips {
```
Comment (maisem) [inline ipn/ipnlocal/c2n.go:309]: wow thats fun

File: tailcfg/tailcfg.go [hunk 3]
```diff
@@ -735,6 +735,7 @@ type Hostinfo struct {
 	GoVersion       string         `json:",omitempty"` // Go version binary was built with
 	RoutableIPs     []netip.Prefix `json:",omitempty"` // set of IP ranges this client can route
 	RequestTags     []string       `json:",omitempty"` // set of ACL tags this node wants to claim
+	WoLMACs         []string       `json:",omitempty"` // MAC addresses to send Wake-on-LAN packets to wake this node (lowercase hex w/ colons)
```
Comment (maisem) [inline tailcfg/tailcfg.go:738]: bump capver too?

**Review by maisem — APPROVED** [review]

File: hostinfo/wol.go (hunk 1 above)
Comment (bradfitz) [reply hostinfo/wol.go:32]: done

**Pull request merged**